Text input nodes ignore shift

**Godot version:**
Commit 72d87cf

**OS/device including version:**
Debian sid

**Issue description:**
Text input nodes ignore shift (and, possibly, other modifier keys), making it impossible to enter shifted characters.

**Steps to reproduce:**
Enter text in the editor. Especially, shifted characters from the upper row like parentheses.

**Minimal reproduction project:**

With this path Shift works, but compilation show this wartning:
```
[ 34%] platform/x11/os_x11.cpp: In member function 'void OS_X11::handle_key_event(XKeyEvent*, bool)':
platform/x11/os_x11.cpp:1770:1: warning: value computed is not used [-Wunused-value]
 + XLookupString(xkeyevent, str, 256, &keysym_unicode, NULL);
```

--- a/platform/x11/os_x11.cpp
+++ b/platform/x11/os_x11.cpp
@@ -1767,7 +1767,8 @@ void OS_X11::handle_key_event(XKeyEvent *p_event, bool p_echo) {
 	XKeyEvent xkeyevent_no_mod = *xkeyevent;
 	xkeyevent_no_mod.state &= ~ShiftMask;
 	xkeyevent_no_mod.state &= ~ControlMask;
-	XLookupString(&xkeyevent_no_mod, str, 256, &keysym_keycode, NULL);
+	XLookupString(xkeyevent, str, 256, &keysym_unicode, NULL);
+	XLookupString(&xkeyevent_no_mod, NULL, 0, &keysym_keycode, NULL);
 
 	// Meanwhile, XLookupString returns keysyms useful for unicode.
 
@@ -1776,8 +1777,6 @@ void OS_X11::handle_key_event(XKeyEvent *p_event, bool p_echo) {
 		xmbstring = (char *)memalloc(sizeof(char) * 8);
 		xmblen = 8;
 	}
-
-	keysym_unicode = keysym_keycode;
 
 	if (xkeyevent->type == KeyPress && xic) {
 

